#!/usr/bin/env python
PACKAGE = "global_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("ignore_path_messages_",	bool_t,   0, "Ignore path messages, only controllable with direct goal",  False)
gen.add("three_point_mode_",   		bool_t,   0, "Send three-point messages instead of position",  False)
gen.add("min_speed_", 				double_t, 0, "Minimum speed for position control",    2.0, 0.0,   3.0)
gen.add("max_speed_", 				double_t, 0, "Maximum speed for position control",    2.0, 1.0,   3.0)
gen.add("three_point_speed_", 		double_t, 0, "Scaling factor for the bezier speed",    1.0, 0.0,   2.0)
gen.add("direct_goal_alt_", 		double_t, 0, "Altitude of direct goal",    2.5, 0.0,   10.0)
gen.add("timeout_startup_", double_t, 0, "After this timeout the companion status is MAV_STATE_CRITICAL", 5, 0, 60)
gen.add("timeout_critical_", double_t, 0, "After this timeout the companion status is MAV_STATE_CRITICAL", 0.5, 0, 10)
gen.add("timeout_termination_", double_t, 0, "After this timeout the companion status is MAV_STATE_FLIGHT_TERMINATION", 15, 0, 1000)

exit(gen.generate(PACKAGE, "global_planner", "PathHandlerNode"))
